servicetalk-http-router-predicate/src/main/java/io/servicetalk/http/router/predicate/HttpPredicateRouterBuilder.java

```diff
@@ -276,17 +275,13 @@ public RouteStarter thenRouteTo(final BlockingStreamingHttpService service) {
             return thenRouteTo0(adapterHolder.adaptor(), adapterHolder.serviceInvocationStrategy());
         }
 
-        private HttpExecutionStrategy serviceOffloads(final Object service) {
-            return null != strategy ? strategy :
-                    service instanceof ExecutionStrategyInfluencer ?
-                            HttpExecutionStrategy.from(((ExecutionStrategyInfluencer) service).requiredOffloads()) :
-                            defaultStrategy();
+        private HttpExecutionStrategy serviceOffloads(final HttpExecutionStrategyInfluencer service) {
+            return null != strategy ? strategy : service.requiredOffloads();
         }
 
-        private RouteStarter thenRouteTo0(final StreamingHttpService route,
-                                          @Nullable final HttpExecutionStrategy routeStrategy) {
+        private RouteStarter thenRouteTo0(final StreamingHttpService route, final HttpExecutionStrategy routeStrategy) {
             assert predicate != null;
-            routes.add(new Route(predicate, route, null == strategy ? null : routeStrategy));
+            routes.add(new Route(predicate, route, routeStrategy));
```

Comment:
This changes the behavior and some tests started to fail, we should keep `null` by default.

Reply:
Sorry, this change was a mistake. reverted.